src/hipercow/dide/mounts.py

```diff
@@ -0,0 +1,118 @@
+import csv
+import platform
+import re
+import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+
+
+@dataclass
+class Mount:
+    host: str
+    remote: Path
+    local: Path
+
+    def __post_init__(self):
+        self.remote = Path(self.remote)
+        self.local = Path(self.local)
+
+
+@dataclass
+class PathMap:
+    path: Path
+    mount: Mount
+    remote: Path
+    relative: Path
+
+
+def remap_path(path: Path, mounts: list[Mount]) -> PathMap:
+    pos = [m for m in mounts if path.is_relative_to(m.local)]
+    if len(pos) > 1:
+        msg = "More than one plausible mount for local directory"
+        raise Exception(msg)
+    elif len(pos) == 0:
+        msg = "Can't map local directory '{path}' to network path"
+        raise Exception(msg)
+    mount = pos[0]
+    relative = path.relative_to(mount.local)
+    if re.match("^[A-Za-z]:/?$", str(mount.local)):
+        remote = mount.local
+    elif mount.host == "qdrive":
+        remote = Path("Q:")
+    else:
+        remote = Path("V:")
+    return PathMap(path, mount, remote, relative)
+
```

Comment:
```suggestion
def safe_platform_system():
  try:
    sysname = platform.system()
  except Exception:
    sysname = "Windows"
  return sysname

```

Reply:
https://docs.python.org/3/library/platform.html#platform.system - this does not suggest that this should throw, much less that throwing means windows! What error do you get here?

Reply:
OK, I see the error on CI - something messy in the mocks I think. I've fixed this another way